src/unix/fsevents.c

```diff
@@ -747,6 +747,8 @@ static void* uv__cf_loop_runner(void* arg) {
                          state->signal_source,
                          *pkCFRunLoopDefaultMode);
 
+  state->loop = NULL;
```

Comment:
we were not resetting this before?

Reply:
It also looks like a race condition to me since it's modified without holding the mutex. `uv__cf_loop_signal()` reads it and runs on the other thread.

Reply:
We weren't—in fact, we should never use or read this field again after this point in the code (we also never branch on it being NULL). The mistake was that we sometimes would (because we didn't hold the necessary mutex). Clearing it here just helps to makes any bad concurrent usage more obvious.